# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: a7b3d464b135
# Full Hash: a7b3d464b135d189f8dc61889a9f3199eb601169
# Author: Robert Mader <robert.mader@posteo.de>
# Date: 2021-04-27 22:18:30
# Regressor Bug: 1707577
# File Overlap Count: 1
# Description:
#   Bug 1707577 - Initialize variables in the constructor in cpp fashion, r=stransky
#   
#   Mostly in order to match code style.
#   Also add some missing variables and remove unused ones.
#   
# ==============================================================================

diff -r b5ba2f343559 -r a7b3d464b135 widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Tue Apr 27 18:57:58 2021 +0000
+++ b/widget/gtk/nsWindow.cpp	Tue Apr 27 18:57:59 2021 +0000
@@ -428,50 +428,72 @@
 #endif
 }
 
-nsWindow::nsWindow() {
-  mIsTopLevel = false;
-  mIsDestroyed = false;
-  mListenForResizes = false;
-  mNeedsDispatchResized = false;
-  mIsShown = false;
-  mNeedsShow = false;
-  mEnabled = true;
-  mCreated = false;
-  mHandleTouchEvent = false;
-  mIsDragPopup = false;
-  mContainer = nullptr;
-  mGdkWindow = nullptr;
-  mShell = nullptr;
-  mCompositorWidgetDelegate = nullptr;
-  mHasMappedToplevel = false;
-  mRetryPointerGrab = false;
+nsWindow::nsWindow()
+    : mIsTopLevel(false),
+      mIsDestroyed(false),
+      mListenForResizes(false),
+      mNeedsDispatchResized(false),
+      mIsShown(false),
+      mNeedsShow(false),
+      mEnabled(true),
+      mCreated(false),
+      mHandleTouchEvent(false),
+      mIsDragPopup(false),
+      mWindowScaleFactorChanged(true),
+      mWindowScaleFactor(1),
+      mCompositedScreen(gdk_screen_is_composited(gdk_screen_get_default())),
+#ifdef MOZ_WAYLAND
+      mNeedsCompositorResume(false),
+      mCompositorInitiallyPaused(false),
+      mNativePointerLockCenter(LayoutDeviceIntPoint()),
+#endif
+      mShell(nullptr),
+      mContainer(nullptr),
+      mGdkWindow(nullptr),
+      mWindowShouldStartDragging(false),
+      mCompositorWidgetDelegate(nullptr),
+      mHasMappedToplevel(false),
+      mRetryPointerGrab(false),
+      mSizeState(nsSizeMode_Normal),
+      mAspectRatio(0.0f),
+      mAspectRatioSaved(0.0f),
+      mLastScrollEventTime(GDK_CURRENT_TIME),
+      mPendingConfigures(0),
+      mGtkWindowDecoration(GTK_DECORATION_NONE),
+      mDrawToContainer(false),
+      mDrawInTitlebar(false),
+      mTitlebarBackdropState(false),
+      mIsPIPWindow(false),
+      mAlwaysOnTop(false),
+      mIsTransparent(false),
+      mTransparencyBitmap(nullptr),
+      mTransparencyBitmapWidth(0),
+      mTransparencyBitmapHeight(0),
+      mTransparencyBitmapForTitlebar(false),
+      mHasAlphaVisual(false),
+      mLastMotionPressure(0),
+      mLastSizeMode(nsSizeMode_Normal),
+      mBoundsAreValid(true),
+      mPreferredPopupRectFlushed(false),
+      mWaitingForMoveToRectCB(false),
+      mPendingSizeRect(LayoutDeviceIntRect(0, 0, 0, 0)),
+#ifdef ACCESSIBILITY
+      mRootAccessible(nullptr),
+#endif
+#ifdef MOZ_X11
+      mXDisplay(nullptr),
+      mXWindow(X11None),
+      mXVisual(nullptr),
+      mXDepth(0),
+#endif
+#ifdef MOZ_WAYLAND
+      mLockedPointer(nullptr),
+      mRelativePointer(nullptr)
+#endif
+{
   mWindowType = eWindowType_child;
-  mSizeState = nsSizeMode_Normal;
-  mBoundsAreValid = true;
-  mAspectRatio = 0.0f;
-  mAspectRatioSaved = 0.0f;
-  mLastSizeMode = nsSizeMode_Normal;
   mSizeConstraints.mMaxSize = GetSafeWindowSize(mSizeConstraints.mMaxSize);
 
-#ifdef MOZ_X11
-  mOldFocusWindow = 0;
-
-  mXDisplay = nullptr;
-  mXWindow = X11None;
-  mXVisual = nullptr;
-  mXDepth = 0;
-#endif /* MOZ_X11 */
-
-#ifdef MOZ_WAYLAND
-  mNeedsCompositorResume = false;
-  mCompositorInitiallyPaused = false;
-  mNativePointerLockCenter = LayoutDeviceIntPoint();
-  mRelativePointer = nullptr;
-  mLockedPointer = nullptr;
-#endif
-  mWaitingForMoveToRectCB = false;
-  mPendingSizeRect = LayoutDeviceIntRect(0, 0, 0, 0);
-
   if (!gGlobalsInitialized) {
     gGlobalsInitialized = true;
 
@@ -489,36 +511,6 @@
     }
 #endif
   }
-
-  mLastMotionPressure = 0;
-
-#ifdef ACCESSIBILITY
-  mRootAccessible = nullptr;
-#endif
-
-  mIsTransparent = false;
-  mTransparencyBitmap = nullptr;
-  mTransparencyBitmapForTitlebar = false;
-
-  mTransparencyBitmapWidth = 0;
-  mTransparencyBitmapHeight = 0;
-
-  mLastScrollEventTime = GDK_CURRENT_TIME;
-
-  mPendingConfigures = 0;
-  mGtkWindowDecoration = GTK_DECORATION_NONE;
-  mDrawToContainer = false;
-  mDrawInTitlebar = false;
-  mTitlebarBackdropState = false;
-
-  mHasAlphaVisual = false;
-  mIsPIPWindow = false;
-  mAlwaysOnTop = false;
-
-  mWindowScaleFactorChanged = true;
-  mWindowScaleFactor = 1;
-
-  mCompositedScreen = gdk_screen_is_composited(gdk_screen_get_default());
 }
 
 nsWindow::~nsWindow() {